src/diffTool/jsonDiff.ts

```diff
@@ -0,0 +1,374 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+/* eslint-disable @typescript-eslint/no-use-before-define */
+import _ from "lodash";
+import { DiffPatcher } from "jsondiffpatch";
+import { ramlToolLogger } from "../logger";
+
+/**
+ * Top level keys/property names in flattened JSON-LD
+ */
+export const JSON_LD_KEY_GRAPH = "@graph";
+export const JSON_LD_KEY_CONTEXT = "@context";
+export const JSON_LD_KEY_ID = "@id";
+
+/**
+ * External property, added by jsondiffpath to indicate that the node property is an array
+ */
+const JSON_DIFF_PATCH_ARRAY_KEY = "_t";
+const JSON_DIFF_PATCH_ARRAY_VALUE = "a";
+
+/**
+ * Difference types that can be identified by jsondiffpath
+ */
+export enum DiffType {
+  ADDED,
+  REMOVED,
+  MODIFIED,
+  TEXT_DIFF,
+  MOVED
+}
+
+/**
+ * Class to hold differences of a JSON node
+ */
+export class NodeDiff {
+  public added?: { [key: string]: any };
+  public removed?: { [key: string]: any };
```

Comment:
I think that `any` here could be replaced with `DiffSignature[]` (see below for definition of `DiffSignature`).

Reply:
The more I am thinking, I feel we can have an interface or type aliases for amf graph and the diffs returned by jsondiff. Lets discuss with the team and take it from there.